### DISCUSSION TITLE
Possible to open pages as modals?

### DISCUSSION THREAD

**Discussion by arladmin** [Budibase/budibase#1199] [Ideas]
Is it possible to designate any page(s) to open in a modal?

Very useful for pages for: 'Create New', 'View Details', 'Edit Details' kind of operations, among others.

**Comment by joebudi** [comment]
This is not possible right now. We don't have a component for this. My immediate thoughts are, does Budibase take care of this for the user, or do we ask the user to add the modal as a component, and create a workflow? Thoughts welcomed.

**Reply by arladmin** [reply]
@joebudi  One option that i use in other platforms, like Knack, is that for each page, a setting is available to enable 'Open as Modal'.

So, you create pages the same way it's currently done, but some pages will just open as a modal, basically, with a little bit of extra css.

**Comment by jekroll555** [comment]
Modals are also good tools for transactional entries. This functionality would be very helpful

**Reply by joebudi** [reply]
I agree. This something we will consider in the next iteration of development. There's a huge piece of work coming up that will involve streamlining the design process, it could involve modals. I will keep you notified.

**Comment by glebgolov-ziosem** [comment]
Has there been any progress for this idea? A modal to peek at details or edit them quickly greatly improves workflows requiring a lot of quick checking or editing; something like the [Bootstrap 5 offcanvas](https://getbootstrap.com/docs/5.0/components/offcanvas/#placement).
